sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetCatalogsOperation.scala

```diff
@@ -68,11 +69,18 @@ private[hive] class SparkGetCatalogsOperation(
       }
       setState(OperationState.FINISHED)
     } catch {
-      case e: HiveSQLException =>
+      case e: Throwable =>
```

Comment:
NonFatal?

Reply:
Hm. I guess we may want to catch a Throwable.
E.g. InterruptedExpression is not catched by NonFatal, and we want to inform the HiveThriftServer2.listener about the error after an interrupt - this definitely can happen in SparkExecuteStatementOperation that is async and can be cancelled. After a ThreadDeath of OutOfMemoryError I think we also want to inform the HiveThriftServer2.listener to not get the query hanging in the UI, as I think the server would continue to go on (I think it won't bring the whole JVM down?).

Reply:
If so, we should list up `InterruptedExpression` here, too? IIUC the reason why we mainly use `NonFatal` in this case is not to catch `NonLocalReturnControl`.  But, yea, this is not my area, so I think @wangyum could suggest more about this.

Reply:
+1 for `Throwable`.

Extractor of non-fatal Throwables. Will not match fatal errors like `VirtualMachineError`.
(for example, `OutOfMemoryError` and `StackOverflowError`, subclasses of `VirtualMachineError`), `ThreadDeath`, `LinkageError`, `InterruptedException`, `ControlThrowable`.

https://github.com/scala/scala/blob/v2.12.10/src/library/scala/util/control/NonFatal.scala#L17-L19